tests/tas-elasticsearch/src/test/java/org/alfresco/elasticsearch/ElasticsearchTests.java

```diff
@@ -0,0 +1,259 @@
+package org.alfresco.elasticsearch;
+
+import org.alfresco.rest.core.RestWrapper;
+import org.alfresco.rest.search.RestRequestQueryModel;
+import org.alfresco.rest.search.SearchNodeModel;
+import org.alfresco.rest.search.SearchRequest;
+import org.alfresco.rest.search.SearchResponse;
+import org.alfresco.utility.Utility;
+import org.alfresco.utility.constants.UserRole;
+import org.alfresco.utility.data.DataContent;
+import org.alfresco.utility.data.DataSite;
+import org.alfresco.utility.data.DataUser;
+import org.alfresco.utility.model.*;
+import org.alfresco.utility.network.ServerHealth;
+import org.alfresco.utility.testrail.ExecutionType;
+import org.alfresco.utility.testrail.annotation.TestRail;
+import org.apache.http.HttpHost;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.elasticsearch.action.get.GetRequest;
+import org.elasticsearch.action.get.GetResponse;
+import org.elasticsearch.client.RequestOptions;
+import org.elasticsearch.client.RestClient;
+import org.elasticsearch.client.RestHighLevelClient;
+import org.elasticsearch.client.indices.CreateIndexRequest;
+import org.elasticsearch.client.indices.GetIndexRequest;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.testng.AbstractTestNGSpringContextTests;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static java.util.Arrays.asList;
+import static org.testng.Assert.*;
+
+@ContextConfiguration("classpath:alfresco-elasticsearch-context.xml")
+/**
+ * In this test we are verifying end-to-end the indexing and search in Elasticsearch.
+ * In order to test ACLs we created 2 sites and 3 users. 
+ */
+public class ElasticsearchTests extends AbstractTestNGSpringContextTests
+{
+    public static final String INDEX_NAME = "alfresco";
+    public static final String FILE_0_NAME = "test.txt";
+    public static final String FILE_1_NAME = "another.txt";
+    public static final String FILE_2_NAME = "user1.txt";
+    public static final String FILE_3_NAME = "user1Old.txt";
+
+    @Autowired
+    public DataUser dataUser;
+    @Autowired
+    public DataContent dataContent;
+    @Autowired
+    public DataSite dataSite;
+
+    @Autowired
+    protected ServerHealth serverHealth;
+
+    @Autowired
+    protected RestWrapper client;
+
+    private UserModel userSite1;
+    private RestHighLevelClient elasticClient;
+    private SiteModel siteModel;
+    private UserModel userSite2;
+    private UserModel userMultiSite;
+    private FileModel sampleContent;
+
+    @BeforeClass(alwaysRun = true)
+    public void dataPreparation()
```

Comment:
Do we need an `@AfterClass` to delete the Index created by these tests?

Reply:
It isn't necessary because the code in the @Before statement deletes the index if it exists and recreates it.
I can add it to avoid leaving the index dirty for other tests.

Reply:
We had a similar problem previously, so my preference is to leave everything clean for other tests.

Reply:
Done!